Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1100,17 +1277,78 @@ class AnimatedProps extends Animated {
   }
 
   __detach(): void {
+    if (this.__isNative && this._animatedView) {
+      this.__disconnectAnimatedView();
+    }
     for (var key in this._props) {
       var value = this._props[key];
       if (value instanceof Animated) {
         value.__removeChild(this);
       }
     }
+    super.__detach();
   }
 
   update(): void {
     this._callback();
   }
+
+  __makeNative(): void {
+    if (!this.__isNative) {
+      this.__isNative = true;
+      for (var key in this._props) {
+        var value = this._props[key];
+        if (value instanceof Animated) {
+          value.__makeNative();
+        }
+      }
+      if (this._animatedView) {
+        this.__connectAnimatedView();
+      }
+    }
+  }
+
+  setNativeView(animatedView: any): void {
+    invariant(this._animatedView === undefined, 'Animated view already set.');
+    this._animatedView = animatedView;
+    if (this.__isNative) {
+      this.__connectAnimatedView();
+    }
+  }
+
+  __connectAnimatedView(): void {
+    invariant(this.__isNative, 'Expected node to be marked as "native"');
+    var nativeViewTag: ?number = findNodeHandle(this._animatedView);
+    invariant(nativeViewTag != null, 'Unable to locate attached view in the native tree');
+    NativeAnimatedAPI.connectAnimatedNodeToView(this.__getNativeTag(), nativeViewTag);
+    if (this._props.style) {
```

Comment:
Instead of doing that, could you change the implementation of __getValue() in the case of natively implemented values to return only the static ones? I'm afraid that if it's handled out of band like this, it can introduce race conditions where the props are updated but the node is not connected anymore and it gets out of sync.

https://github.com/facebook/react-native/pull/6466/files#diff-f048d92ca0be679bc38d38147b311100R1423